### PR TITLE
#230 Expose settings for bigNumber/longFile modes on tar file

### PR DESCRIPTION
**Pull request by kevinconaway**
Fixes for #230.

I wasn't sure of the best way to add test cases for this.  There does not seem to be a way to verify the modes on the resulting tar file.  I did test it locally to ensure it works.

### CODE REVIEW

**Comment by kevinconaway** [comment]
Not sure why the appveyor check failed, it seems like its an environmental issue because it didn't seem to start

https://ci.appveyor.com/project/tcurdt/jdeb/build/1.0.18

File: src/main/java/org/vafer/jdeb/DataBuilder.java [hunk 1]
```diff
@@ -78,21 +78,27 @@ private void checkField(String name, int length) throws IOException {
...
 
-        final TarArchiveOutputStream tarOutputStream = new TarArchiveOutputStream(compression.toCompressedOutputStream(new FileOutputStream(output)));
-        tarOutputStream.setLongFileMode(TarArchiveOutputStream.LONGFILE_GNU);
+        final TarArchiveOutputStream tarOutputStream = new TarArchiveOutputStream(
+            options.compression().toCompressedOutputStream(new FileOutputStream(output))
+        );
+        tarOutputStream.setLongFileMode(options.longFileMode());
+        tarOutputStream.setBigNumberMode(options.bigNumberMode());
+
+        System.out.println("##### longFileMode: " + options.longFileMode());
```
Comment (tcurdt) [inline src/main/java/org/vafer/jdeb/DataBuilder.java]: Those `println` should be removed.

**Comment by tcurdt** [comment]
Don't worry about appveyor - still not setup properly.
Thanks for the contribution!

File: src/main/java/org/vafer/jdeb/DataBuilder.java (hunk 1 above)
Comment (kevinconaway) [inline src/main/java/org/vafer/jdeb/DataBuilder.java]: Indeed they should.  Gone.

**Comment by tcurdt** [comment]
What I am wondering though is whether it would make sense to just always build with big number support. Would be good to check what the official packages have.

**Comment by kevinconaway** [comment]
> What I am wondering though is whether it would make sense to just always build with big number support. Would be good to check what the official packages have.

Good point.  The maven-assembly-plugin exposes the _tarLongFileMode_ property but does not expose anything about the big number mode.

Looking at the code, it seems that they drive the bigNumber mode depending on what the _tarLongFileMode_ is
https://github.com/sonatype/plexus-archiver/blob/master/src/main/java/org/codehaus/plexus/archiver/tar/TarArchiver.java#L141

**Comment by kevinconaway** [comment]
@tcurdt what are your thoughts on the above?  I think having both as properties makes things more flexible but that flexibility may be unwarranted

**Comment by tcurdt** [comment]
@kevinconaway Not sure what the spec says about this. What I would do though is to first check what the native deb packages use.

If we could away with not adding such low-level options I'd be all for it.

**Comment by kevinconaway** [comment]
After looking in to it a bit, it seems we should just default it to POSIX format for tars.  You can read a bit about gnu format vs posix format here:

https://www.gnu.org/software/tar/manual/html_chapter/tar_8.html

Seems like all modern implementations of tar will be able to read the POSIX format headers and apply it.

It _might_ be a good idea to leave the options in in case there is a good use case for using another format but thats your call.

If you're happy with the above, I'll change the PR to default both settings to POSIX

**Comment by tcurdt** [comment]
Sounds good! Thanks for digging into this!

**Comment by kevinconaway** [comment]
Pushed changes per our discussion

**Comment by tcurdt** [comment]
Thanks!

**Comment by tcurdt** [comment]
Urgh - seems that broke the build for jdk8 :-/

https://travis-ci.org/tcurdt/jdeb/jobs/107977245

**Comment by tcurdt** [comment]
Seems like `org.vafer.jdeb.ant.DebAntTaskTestCase#testPackageConffiles` doesn't like the changes.

**Comment by tcurdt** [comment]
Nevermind - seems that was a CI glitch.

**Comment by kevinconaway** [comment]
@tcurdt Sorry but I was mistaken about the POSIX compatible tar files working with dpkg.

See the following discussion: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=291194

It appears that dpkg will fail intermittently, not all the time, which is why it initially passed my local testing.  Further follow up testing caused it to fail though.

I've updated my branch to default to GNU for longfile/bigNumber mode.

**Comment by tcurdt** [comment]
Indeed - that doesn't look good. Can you please create another PR?

**Pull request merged**